python/taichi/lang/impl.py

```diff
@@ -490,7 +490,13 @@ def create_field_member(dtype, name):
     dtype = cook_dtype(dtype)
 
     # primal
-    x = Expr(get_runtime().prog.make_id_expr(""))
+    try:
+        x = Expr(get_runtime().prog.make_id_expr(""))
+    except:
+        raise TaichiRuntimeError(
```

Comment:
This fix is a bit hacky, would you mind explaining  which issue are you trying to solve and what was the original error message?

Reply:
Yea then the right way is to check if `get_runtime().prog is None` instead of this `try...except` block?

Reply:
Yes that's a better solution.